### ISSUE TITLE
aspnetcore-engine@5.0.0-beta.1 not working with TransferStateModule 

### ISSUE THREAD

**Issue by naveedahmed1** [angular/universal#832]
* **I'm submitting a ...**
```
- [ x] bug report
- [ ] feature request
- [ ] support request => Please do not submit support request here, see note at the top of this template.
```

* **What modules are related to this Issue?**
```
- [x ] aspnetcore-engine
- [ ] express-engine
- [ ] hapi-engine
```

* **Do you want to request a *feature* or report a *bug*?**
It seems to be a bug.

After upgrading to aspnetcore-engine@5.0.0-beta.1 I tried adding ServerTransferStateModule and BrowserTransferStateModule from @angular/platform-server' and @angular/platform-browser respectively. As suggested https://next.angular.io/api/platform-server/ServerTransferStateModule and  https://medium.com/@evertonrobertoauler/angular-5-universal-with-transfer-state-using-angular-cli-19fe1e1d352c

But after this change, I am now receiving error `Error: The selector "app-root" did not match any elements` and no markup is generated at server.

* **Please tell us about your environment:**

- Angular version: 5.0.0-rc.1
- Browser: [Chrome]
- Language: [TypeScript]
- OS:  [Windows]
- Platform: [.NETCore]

**Comment by vikerman** [comment]
@MarkPieszak - Looks like the DOM nodes added by TransferState is affecting the ASP.NET engine parsing?

**Comment by naveedahmed1** [comment]
@MarkPieszak I think with new ASP.Net engine the old TransferStateModule approach that we followed in https://github.com/MarkPieszak/aspnetcore-angular2-universal is also broken transferData is always empty. Can this please be fixed any time soon?

**Comment by naveedahmed1** [comment]
Is there any update on this?

**Comment by naveedahmed1** [comment]
Any update on this? Seems like I am the only person using this engine :D

**Comment by daveriedstra** [comment]
we're using it too, @naveedahmed1, and looking forward to an update

**Comment by naveedahmed1** [comment]
@daveriedstra try new upcoming template from asp.net team you wont be disappointed https://github.com/aspnet/JavaScriptServices/issues/1288#issuecomment-346003334

I think Universal team will ultimately retire `aspnetcore-engine` after the new template is release.

**Comment by devel112** [comment]
aspnetcore-engine is taking elements out of the document (style, scripts, links) so that .NET can paste the elements where it wishes, the problem is that it's only getting scripts from head, and servertransfersatemodule is appending it's state script to body.

Is it that wise to rip appart the dom and recreating it in .NET? From this example it seems risky. 
If i'm not wrong this was done because of better SEO, but couldn't atleast renderModuleFactory give back SEO friendly (i mean, the function is created by Google afterall 😸) html?

**Comment by devel112** [comment]
Fixed in #889
